src/chainstate/burn/db/sortdb.rs

```diff
@@ -1755,30 +1755,29 @@ impl<'a> SortitionHandleConn<'a> {
                     warn!("Missing parent"; "burn_header_hash" => %prepare_end_bhh);
                     BurnchainError::MissingParentBlock
                 })?;
-        let my_height = SortitionDB::get_block_height(self.deref(), &prepare_end_sortid)?
+        let block_height = SortitionDB::get_block_height(self.deref(), &prepare_end_sortid)?
             .expect("CORRUPTION: SortitionID known, but no block height in SQL store");
 
         // if this block is the _end_ of a prepare phase,
-        if !(my_height % pox_consts.reward_cycle_length == 0) {
-            test_debug!(
-                "My height = {}, reward cycle length == {}",
-                my_height,
-                pox_consts.reward_cycle_length
+        let effective_height = block_height - self.context.first_block_height as u32;
+        let position_in_cycle = effective_height % pox_consts.reward_cycle_length;
+        if position_in_cycle != 0 {
+            debug!(
+                "effective_height = {}, reward cycle length == {}",
+                effective_height, pox_consts.reward_cycle_length
             );
             return Err(CoordinatorError::NotPrepareEndBlock);
         }
 
-        let prepare_end = my_height;
-        // if this block isn't greater than prepare_length, then this shouldn't be the end of a prepare block
-        if prepare_end < pox_consts.prepare_length {
-            test_debug!(
-                "prepare_end = {}, pox_consts.prepare_length = {}",
-                prepare_end,
-                pox_consts.prepare_length
+        if effective_height == 0 {
```

Comment:
Wait, wasn't this `!(effective_height == 0)` before? Shouldn't this be `effective_height != 0`?